ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/Sink.java

```diff
@@ -53,30 +55,35 @@ public static void main(String[] args) {
           Env.getLong(BATCH_MAX_BYTES, 100_000_000L), // default 100MB
           Env.getInt(BATCH_MAX_MESSAGES, 1_000_000), // default 1M messages
           Env.getDuration(BATCH_MAX_DELAY, "10m"), // default 10 minutes
-          gcsPrefix, format);
+          gcsPrefix, format)::apply;
       // These limits are higher for the GCS output because it has higher batch latency
       // default 1M messages
       maxOutstandingElementCount = Env.getLong(MAX_OUTSTANDING_ELEMENT_COUNT, 1_000_000L);
       // default 1GB
       maxOutstandingRequestBytes = Env.getLong(MAX_OUTSTANDING_REQUEST_BYTES, 1_000_000_000L);
     } else {
-      output = new BigQuery.Write(BigQueryOptions.getDefaultInstance().getService(),
-          // BigQuery.Write.Batch.getByteSize reports protobuf size, which can be ~1/3rd more
-          // efficient than the JSON that actually gets sent over HTTP, so we use to 60% of the 10MB
-          // API limit by default.
-          Env.getLong(BATCH_MAX_BYTES, 6_000_000L), // default 6MB
-          // BigQuery Streaming API Limits maximum rows per request to 10,000
-          Env.getInt(BATCH_MAX_MESSAGES, 10_000), // default 10K messages
-          Env.getDuration(BATCH_MAX_DELAY, "1s"), // default 1 second
-          Env.getString(OUTPUT_TABLE), format);
+      if (Env.optString(OUTPUT_TOPIC).isPresent()) {
```

Comment:
I think this should be an else/if rather than nested. 

But perhaps we should really have a bit of a refactor here. We should validate that only one of (OUTPUT_BUCKET, OUTPUT_TOPIC, OUTPUT_TABLE) is specified in the environment. Perhaps this warrants a new method in `Env` and factoring out a `buildOutput` method:

```
// Throws IllegalArgumentException if zero or more than one of these variables is set.
String outputVarName = Env.getMutuallyExclusiveOption(ImmutableList.of(OUTPUT_BUCKET, OUTPUT_TOPIC, OUTPUT_TABLE));

output = buildOutput(outputVarName);

...

public Function<PubsubMessage, CompletableFuture<?>> buildOutput(String outputVarName) {

switch (outputVarName) {
  case OUTPUT_BUCKET:
    return ...;
  ...
}
```

Reply:
I'll file a follow up PR to refactor this